File: dotnet/src/Microsoft.Agents.Protocol.Validation/Validators/ContentValidators/ImageContentValidator.cs
using Microsoft.Agents;

namespace Microsoft.Agents.Protocol.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for ImageContent.
/// </summary>
public class ImageContentValidator : ContentValidatorBase<ImageContent>
{
    public override ValidationResult Validate(ImageContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-006: ImageContent must have uri OR data (at least one)
        if (string.IsNullOrWhiteSpace(content.Uri))
        {
            errors.Add(CreateError(
                ValidationErrorCode.CNT_006,
                "ImageContent must have a uri",
                "Uri"));
        }

        // CNT-007: ImageContent width/height must be positive integers
        var widthError = ValidatePositive(content.Width, "Width",
            ValidationErrorCode.CNT_007,
            "ImageContent width must be a positive integer");
        if (widthError != null)
            errors.Add(widthError);

        var heightError = ValidatePositive(content.Height, "Height",
            ValidationErrorCode.CNT_007,
            "ImageContent height must be a positive integer");
        if (heightError != null)
            errors.Add(heightError);

        // CNT-008: ImageContent mime-type must be valid (image/*)
        if (!string.IsNullOrWhiteSpace(content.MimeType) &&
            !content.MimeType.StartsWith("image/", StringComparison.OrdinalIgnoreCase))
        {
            errors.Add(CreateError(
                ValidationErrorCode.CNT_008,
                $"ImageContent mime-type '{content.MimeType}' must start with 'image/'",
                "MimeType"));
        }

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/ImageContentValidator.ts
import type { ImageContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';
import { ValidationError } from '../../core/ValidationError';
import { ValidationErrorCode } from '../../core/ValidationErrorCode';

/**
 * Validator for ImageContent.
 */
export class ImageContentValidator extends ContentValidatorBase<ImageContent> {
  public validate(content: ImageContent, context?: ValidationContext): ValidationResult {
    const errors: ValidationError[] = [];

    // CNT-006: must have uri
    const uriError = this.validateNotEmpty(
      content.uri,
      'uri',
      ValidationErrorCode.CNT_006,
      'ImageContent must have uri'
    );
    if (uriError) {
      errors.push(uriError);
    }

    // Validate URI format (if present)
    if (content.uri) {
      const uriFormatError = this.validateUri(
        content.uri,
        'uri',
        ValidationErrorCode.CNT_006,
        'ImageContent uri must be a valid URI'
      );
      if (uriFormatError) {
        errors.push(uriFormatError);
      }
    }

    // CNT-007: width and height must be positive (if present)
    const widthError = this.validatePositive(
      content.width,
      'width',
      ValidationErrorCode.CNT_007,
      'ImageContent width must be positive'
    );
    if (widthError) {
      errors.push(widthError);
    }

    const heightError = this.validatePositive(
      content.height,
      'height',
      ValidationErrorCode.CNT_007,
      'ImageContent height must be positive'
    );
    if (heightError) {
      errors.push(heightError);
    }

    // CNT-008: mime-type must be image/* (if present)
    if (content.mimeType && !content.mimeType.toLowerCase().startsWith('image/')) {
      errors.push(
        new ValidationError(
          ValidationErrorCode.CNT_008,
          'ImageContent mime-type must start with "image/"',
          'mimeType'
        )
      );
    }

    return new ValidationResult(errors);
  }
}
